libs/apifaker/src/main/java/com/woocommerce/android/apifaker/EndpointExportManager.kt

```diff
@@ -0,0 +1,58 @@
+package com.woocommerce.android.apifaker
+
+import android.content.Context
+import android.net.Uri
+import com.google.gson.Gson
+import com.woocommerce.android.apifaker.db.EndpointDao
+import com.woocommerce.android.apifaker.models.MockedEndpoint
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+internal class EndpointExportManager @Inject constructor(
+    private val gson: Gson,
+    private val endpointDao: EndpointDao,
+    private val context: Context
+) {
+    suspend fun exportEndpoints(
+        endpoints: List<MockedEndpoint>,
+        uri: Uri
+    ): Result<Unit> {
+        return withContext(Dispatchers.IO) {
+            val outputStream = context.contentResolver.openOutputStream(uri)
+            if (outputStream == null) {
+                return@withContext Result.failure(IllegalStateException("Could not open output stream for uri: $uri"))
+            }
```

Comment:
Minor np, this could be simplified to use elvis operator

```suggestion
                ?: return@withContext Result.failure(IllegalStateException("Could not open output stream for uri: $uri"))
```

Reply:
Thanks for the suggestion 👍, I applied it in 58876ca